### PR TITLE
Fast update

### PR DESCRIPTION
**Pull request by ibgreen**

### CODE REVIEW

**Comment by ibgreen** [comment]
* Fix for spurious react render of identical layers when using react-map-gl 3.0.0-alpha.12 (avoids crashing deck.gl)
* Log tuning. 
* Some light code cleanup
* Move long comments in code into README.

File: src/lib/layer-manager.js [hunk 1]
```diff
@@ -114,9 +90,14 @@ export default class LayerManager {
   }
 
   updateLayers({newLayers}) {
-    /* eslint-disable */
-    assert(this.context.viewport,
-      'LayerManager.updateLayers: viewport not set');
+    // TODO - something is generating state updates that cause rerender of the same
+    if (newLayers === this.lastRenderedLayers) {
```
Comment (shaojingli) [inline src/lib/layer-manager.js]: I'm a bit confused over this newly added `lastRenderedLayers`... we probably should have a doc explaining the relations between `this.prevLayers`, `this.lastRenderedLayers`, `this.layers`, etc... (and also additional variables like `generatedLayers`, `oldLayers`, `newLayers`)

Comment (ibgreen) [inline src/lib/layer-manager.js]: This is a hack to avoid a crash caused by react-map-gl v3.

Comment (Pessimistress) [inline src/lib/layer-manager.js]: I am not able to reproduce this on master.

**Pull request merged**